domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/induction/service/InductionEventService.kt

```diff
@@ -10,10 +10,10 @@ interface InductionEventService {
   /**
    * Implementations providing custom code for when a [Induction] is created.
    */
-  fun inductionCreated(createdInduction: Induction)
+  suspend fun inductionCreated(createdInduction: Induction)
```

Comment:
Having to declare the interface method sigs as `suspend` here in the domain feels slightly dirty to me. It feels like an implementation detail is bleeding into the domain, and our domain is meant to be "pure" / "clean" making no framework etc demands on the implementation.

(I accept its a kotlin project and environment, so perhaps not a major issue, but you can see the effect this method sig change has on the java bytecode in `UpdateGoalRequestReferenceConstraintValidatorTest` below 👇 )

Reply:
"This is the way". Kotlin is all about bytecode manipulation I'm afraid. In essence the big win is that converting to async code is as simple as adding the suspend keyword then using async/launch in the method. If you were to try to achieve the same outcomes using Java futures you'd maybe come to appreciate the elegance of Kotlin suspend functions.